gitter.py

```diff
@@ -214,7 +214,7 @@ class GitterBackend(ErrBot):
     This is the Gitter backend for errbot.
     """
 
-    def __init__(self, config):
+    def __init__(self, config, joined_rooms=[]):
```

Comment:
Mutable arguments in a function definition are [a no-no in Python](http://python-guide-pt-br.readthedocs.io/en/latest/writing/gotchas/#mutable-default-arguments).

Also, since a bot starts out not joined to any room, there's no need to make this settable via the constructor. So you leave this off, and just do `self._joined_rooms = []` in the body of the function.